pkg/apis/api.acorn.io/v1/app.go

```diff
@@ -0,0 +1,132 @@
+package v1
+
+import (
+	v1 "github.com/acorn-io/runtime/pkg/apis/internal.acorn.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type App struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   v1.AppInstanceSpec `json:"spec,omitempty"`
+	Status AppStatus          `json:"status,omitempty"`
+}
+
+func (in *App) GetStopped() bool {
+	return in.Spec.Stop != nil && *in.Spec.Stop && in.DeletionTimestamp.IsZero()
+}
+
+func (in *App) GetRegion() string {
+	if in.Spec.Region != "" {
+		return in.Spec.Region
+	}
+	return in.Status.Defaults.Region
+}
+
+type AppStatus struct {
```

Comment:
Is it possible to embed `AppStatus` into `AppInstanceStatus`? This would make maintenance and conversion (below) much simpler.

Reply:
good idea, I'll give it a try.

Reply:
done